core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/samModels.scala

```diff
@@ -24,7 +24,7 @@ object SamResourceId {
     override def resourceType: SamResourceType = SamResourceType.Project
   }
 
-  final case class AppSamResourceId(resourceId: String, accessScope: Option[AppAccessScope]) extends SamResourceId {
+  final case class AppSamResourceId(resourceId: String)(accessScope: Option[AppAccessScope]) extends SamResourceId {
```

Comment:
This is not strictly relevant to hail batch, but is to fix a bug when testing workspace-shared apps in Leo.

Reply:
Oh yeah, I definitely introduced this, I would love some more context if you can share 🙏

Reply:
Yup, just need to page it back into memory. 😵‍💫 

This came up because I was trying to spin up a shared hail-batch app for the Hail team to use for testing, and noticed some strangeness with Terra UI calling `listAppsV2`. But it's a _little_ out of scope because we're not really planning on making Hail Batch a shared app, for the MVP anyway.

It relates to the algorithm we're using for `listApps`:
```
get apps from Leo DB
get user-visible apps from Sam, filtered by policy
take intersection of above 2 sets
```

In step 3 we're relying on equality of the `AppSamResourceId` type. In Leo we have the `accessScope` field. But in the Sam deserialization it's set to `None`, because the Sam model doesn't really have a concept of access scope. So the equality check was failing, and `listAppsV2` was returning `[]` even though I had a shared app.

The above fix is a Scala trick to make `accessScope` a separate parameter list, which means it isn't included in the case class `equals()` method. I suppose I could have also changed the Sam deserialization logic to set `accessScope` if the resource type is `kubernetes-app-shared`. 

In hindsight I think it would have been better to make a separate type `case class SharedAppSamResourceId(resourceId: String)` instead of adding the `accessScope` field to `AppSamResourceId`. It would have avoided weird errors like this, really hard to predict that in advance though. And _really_, I think we should work toward just rewriting the Leo-Sam integration using the generated Sam Java client, which is probably less error prone and easier to understand.

Reply:
Oh cool! That sounds good then, can leave it this way in this PR since it is working :)

Reply:
I actually found (I think) a more natural way of doing this, so resolving this comment for now.